components/eam/src/physics/crm/rrtmgp/radiation.F90

```diff
@@ -1387,9 +1403,16 @@ subroutine radiation_tend(state_in,ptend,    pbuf,          cam_out, cam_in,  &
             ! shortwave and longwave.
             if (radiation_do('sw') .or. radiation_do('lw')) then
 
-               ! Calculate effective radius for optics used with single-moment microphysics
-               if (use_MMF .and. (trim(MMF_microphysics_scheme) .eq. 'sam1mom')) then 
+               ! Calculate effective radius for optics used with single-moment microphysics and for COSP
+               if (use_MMF) then 
                   call cldefr(state%lchnk, ncol, state%t, rel, rei, state%ps, state%pmid, landfrac, icefrac, snowh)
+                  do iz = 1,crm_nz
+                     do ic = 1,ncol
+                        ilay = pver - iz + 1
+                        crm_rel(ic,ix,iy,iz) = rel(ic,ilay)
+                        crm_rei(ic,ix,iy,iz) = rei(ic,ilay)
+                     end do
+                  end do
```

Comment:
Just noticed an error here from my test run last night. We're missing the loops for ix and iy.

Reply:
Yup good call, this section was moved in a recent commit (and isn't actually used by the simulators we are looking at). Fixed now.